canteenWeb/models.py

```diff
@@ -7,15 +7,35 @@
 
 
 class User(AbstractUser):
-    pass
+    USERNAME_FIELD = models.CharField(max_length=30, unique=True, default="")  # sap id
+    EMAIL_FIELD = models.EmailField(max_length=40, blank=True, unique=True)
+    first_name = models.CharField(max_length=15)
+    last_name = models.CharField(max_length=15)
```

Comment:
Remove USERNAME_FIELD, EMAIL_FIELD, first_name, and last_name

Reply:
Why ? USERNAME_FIELD is for sap id right as we discussed.Give me the reason why you want to remove it .

Reply:
That I wrote with respect to AbstractBaseUser model. No need in AbstractUser model. Here are the differences: https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html